iroh/src/discovery/pkarr.rs

```diff
@@ -255,9 +258,9 @@ impl PublisherService {
             }
             // Wait until either the retry/republish timeout is reached, or the node info changed.
             tokio::select! {
-                res = self.watcher.watch_async() => match res {
-                    Ok(()) => debug!("Publish node info to pkarr (info changed)"),
-                    Err(_disconnected) => break,
+                res = self.watcher.updated() => match res {
```

Comment:
I hope this is cancel safe ;)

Reply:
Yes. I've added comments & a rough test.
Really the only thing that can go wrong here is that the watcher ends up in a broken state after cancellation.
That is not the case.

Missing values is expected of the interface anyways.